lib/browser/api/net.js

```diff
@@ -255,7 +255,7 @@ class ClientRequest extends EventEmitter {
 
     const key = name.toLowerCase()
     this.extraHeaders[key] = value
-    this.urlRequest.setExtraHeader(name, value)
+    this.urlRequest.setExtraHeader(name, value.toString())
```

Comment:
Looking at line 249, we only check against undefined, which means `null` could get through to here and the error would be a bit confusing, maybe the `value === undefined` check could be updated to `value == null` since that is the check we typically use in lots of places. What do you think?